tests/baselines/reference/mappedTypeIndexedAccess.errors.txt

```diff
@@ -36,7 +36,7 @@ tests/cases/compiler/mappedTypeIndexedAccess.ts(24,5): error TS2322: Type '{ key
     // Error expected here
     let pair2: Pairs<FooBar>[keyof FooBar] = {
         ~~~~~
-!!! error TS2322: Type '{ key: "foo"; value: number; }' is not assignable to type 'Pair<FooBar>'.
+!!! error TS2322: Type '{ key: "foo"; value: number; }' is not assignable to type '{ key: "foo"; value: string; } | { key: "bar"; value: number; }'.

```

Comment:
Arguable in this specific case.

Reply:
Yeah, we now need deliberate logic to "get back to the same union", so to speak. Previously it would happen automatically because we'd (non-deterministically) attach an alias to the union and then use it _everywhere_. In this particular case we could go either way, but we need additional logic to recognize that the two operations neutralize each other and produce the original union.